src/client/game.cpp

```diff
@@ -2439,6 +2442,17 @@ void Game::toggleAutoforward()
 		m_game_ui->showTranslatedStatusText("Automatic forward disabled");
 }
 
+void Game::toggleSneakLock()
+{
+	bool sneak_lock = !g_settings->getBool("sneak_lock");
```

Comment:
Why do you think global, persistent settings are the right place for this? I would not expect this to be persisted.

It seems to me that `PlayerSettings` (see `localplayer.h`), as used by autoforward, would be more appropriate.

Reply:
Toggle fly is persisted globally iirc.

Reply:
so is continuous_forward

Reply:
Indeed `PlayerSettings` are currently persisted (which I'm not a big fan of, but it's fine if you are consistent with this).

But still, I think your implementation should go through `PlayerSettings` for consistency with the other player settings like autoforward. This makes it easier to change this in bulk, e.g. if we were to move them to a different place in the future.